src/keypair.rs

```diff
@@ -95,36 +102,6 @@ trait KeyPairExt: KeyPair + Debug {
 #[cfg(test)]
 impl<KP> KeyPairExt for KP where KP: KeyPair + Debug {}
 
-/// This is a blanket implementation of SizedBytes for any instance of KeyPair
-/// with any length of keys. This encodes that both keys have the same size,
-/// and that we serialize the public key first, followed by the private key in
-/// binary formats (and expect it in this order upon decoding).
-impl<T, KP> SizedBytes for KP
-where
-    T: SizedBytes + Clone + for<'a> TryFrom<&'a [u8], Error = InternalPakeError>,
-    KP: KeyPair<Repr = T> + PartialEq,
-    T::Len: Add<T::Len>,
-    Sum<T::Len, T::Len>: ArrayLength<u8>,
-    Sum<T::Len, T::Len>: Sub<T::Len, Output = T::Len>,
-    Diff<Sum<T::Len, T::Len>, T::Len>: ArrayLength<u8>,
-{
-    type Len = Sum<T::Len, T::Len>;
-
-    fn to_arr(&self) -> GenericArray<u8, Self::Len> {
-        let private = self.private().to_arr();
-        let public = self.public().to_arr();
-        public.concat(private)
-    }
-
-    fn from_arr(arr: &GenericArray<u8, Self::Len>) -> Result<Self, InternalPakeError> {
-        let (public_key, private_key): (GenericArray<u8, T::Len>, GenericArray<u8, _>) =
-            GenericArray::split(arr.clone());
-        let public = <T as SizedBytes>::from_arr(&public_key)?;
-        let private = <T as SizedBytes>::from_arr(&private_key)?;
-        KP::new(public, private)
-    }
-}
-
```

Comment:
@kevinlewi  This part is a really nice removal I think, replaced by the 1-line derivation of `SizedBytes` on `X25519KeyPair` on line 131.